File: Pythonleetcode/Leetcode202.py
class Solution(object):
    def isHappy(self, n):

        def back(n):
            m=0
            while n!=0:
                m=m+(n%10)**2
                n=n//10
            return m

        his=[n]
        while True:
            if n==1:
                return True

            n=back(n)



            if n in his:
                return False
            elif n==1:
                return True
            his.append(n)


print(Solution().isHappy(19))

File: CppPractice/Project1/Project1/Leetcode202.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    bool isHappy(int n) {
        int temp_sum = 0;
        int temp_n = n;
        int remain;
        vector<int> exist;
        while (n > 0)
        {
            // cout << n <<"  "; 

            while (temp_n > 0)
            {
                remain = temp_n % 10;
                temp_n = temp_n / 10;
                temp_sum += pow(remain, 2);

            }
            if (temp_sum == 1)
            {
                return true;
            }
            else if (find(exist.begin(), exist.end(), temp_sum) != exist.end())
            {



                return false;

            }
            exist.push_back(temp_sum);

            n = temp_sum;

            temp_sum = 0;
            temp_n = n;
        }
        return false;
    }
};